shared/tests/CMakeLists.txt

```diff
@@ -1,19 +1,9 @@
-# This directory contains our message types and other shared types
-include_directories(${CMAKE_SOURCE_DIR}/shared/)
-
-include_directories(${NUCLEAR_MESSAGE_INCLUDE_DIRS})
-include_directories(${NUCLEAR_UTILITY_INCLUDE_DIRS})
-include_directories(${NUCLEAR_EXTENSION_INCLUDE_DIRS})
-
 # This gets all of our source files
 file(GLOB_RECURSE test_src "*/**.cpp" "*/**.hpp")
 
 if(test_src)
   add_executable(TestShared "${CMAKE_SOURCE_DIR}/${NUCLEAR_SHARED_DIR}/tests/test.cpp" ${test_src})
-  target_link_libraries(
-    TestShared ${NUCLEAR_ADDITIONAL_SHARED_LIBRARIES} ${NUCLEAR_UTILITY_LIBRARIES} ${NUCLEAR_MESSAGE_LIBRARIES}
-    ${NUClear_LIBRARIES} ${NUCLEAR_TEST_LIBRARIES}
-  )
+  target_link_libraries(TestShared nuclear::message nuclear::utility)
```

Comment:
```suggestion
  target_link_libraries(TestShared PUBLIC nuclear::message nuclear::utility)
```

Reply:
Technically I think this could also be PRIVATE as TestShared doesn't expose any functionality that comes from nuclear::message or nuclear::utility in it's headers